checker/src/org/checkerframework/checker/regex/RegexQualifiedTransfer.java

```diff
@@ -45,82 +59,218 @@ public RegexQualifiedTransfer(QualAnalysis<QualParams<Regex>> analysis) {
+                    new ConditionalTransferResult<>(resultIn.getResultValue(), thenStore, elseStore);
+            FlowExpressionContext context = FlowExpressionParseUtil.buildFlowExprContextForUse(n,
+                    analysis.getContext());
+            Receiver firstParam;
+            try {
+                // TODO: is this the easiest way to do this?
+                firstParam = FlowExpressionParseUtil.parse("#1", context,
+                        analysis.getContext().getTypeFactory().getPath(n.getTree()));
+            } catch (FlowExpressionParseException e) {
+                firstParam = null;
+                assert false;
+            }
+
+            // add annotation with correct group count (if possible,
+            // regex annotation without count otherwise)
+            Node count = n.getArgument(1);
+            int groupCount;
+            if (count instanceof IntegerLiteralNode) {
+                IntegerLiteralNode iln = (IntegerLiteralNode) count;
+                groupCount = iln.getValue();
+            } else {
+                groupCount = 0;
+            }
+            Regex regex = new RegexVal(groupCount);
+            thenStore.insertValue(firstParam, new QualParams<>(new GroundQual<>(regex)));
+            return newResult;
+        } else if (ElementUtils.matchesElement(method, AS_REGEX_METHOD_NAME, String.class, int.class)) {
+            // RegexUtil.asRegex(s, groups) method
+            // (No special case is needed for asRegex(String) because of
+            // the annotation on that method's definition.)
+
+            // add annotation with correct group count (if possible,
+            // regex annotation without count otherwise)
+            Node count = n.getArgument(1);
+            int groupCount;
+            if (count instanceof IntegerLiteralNode) {
+                IntegerLiteralNode iln = (IntegerLiteralNode) count;
+                groupCount = iln.getValue();
+            } else {
+                groupCount = 0;
+            }
+            QualParams<Regex> regex = new QualParams<>(new GroundQual<Regex>(new RegexVal(groupCount)));
+            QualValue<QualParams<Regex>> newResultValue = analysis.createSingleAnnotationValue(regex,
+                    resultIn.getResultValue().getType().getUnderlyingType().getOriginalType());
+            return new RegularTransferResult<>(newResultValue, resultIn.getRegularStore());
+        }
+        return resultIn;
+    }
+
     /**
      * Returns true if the given receiver is a class named "RegexUtil".
      */
     private boolean isRegexUtil(String receiver) {
         return receiver.equals("RegexUtil") || receiver.endsWith(".RegexUtil");
     }
+
+
+    @Override
+    public
+    TransferResult<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>>
+    visitLessThan(LessThanNode n,
+            TransferInput<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>> in) {
+        // Look for: constant < mat.groupCount()
```

Comment:
These are useful comments.  Could you make them into Javadoc (together with an {@inheritDoc} Javadoc directive), here and in the following methods?